### ISSUE TITLE
macOS `3.0.0rc1` segmentation fault when training

### ISSUE THREAD

**Issue by mikeumo** [DeepLabCut/DeepLabCut#2634]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

I label videos and create a training dataset. however, when I click "train network" the GUI (and python) crashes. I get:

```
Loading pretrained weights from Hugging Face hub (timm/resnet50_gn.a1h_in1k)
[timm/resnet50_gn.a1h_in1k] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
zsh: segmentation fault  python -m deeplabcut
```

Can you help me figure it out?

### Operating System

MacOS Sonoma 14.5

### DeepLabCut version

dlc version: 3.0.0 rc1

### DeepLabCut mode

single animal

### Device type

gpu

### Steps To Reproduce

I follow instructions from https://github.com/DeepLabCut/DeepLabCut/pull/2613
I do not install CUDA. 

Just follow instructions in the 2018 paper and the suggested workflow

my config:
```
# Project definitions (do not edit)
Task: RatTest
scorer: Mike
date: Jun21
multianimalproject: false
identity:


# Project path (change when moving around)
project_path: .../RatTest-Mike-2024-06-21


# Default DeepLabCut engine to use for shuffle creation (either pytorch or tensorflow)
engine: pytorch


# Annotation data set configuration (and individual video cropping parameters)
video_sets:
<removed>,
bodyparts:
- bodypart1
- bodypart2
- bodypart3
- objectA

# Fraction of video to start/stop when extracting frames for labeling/refinement
start: 0
stop: 1
numframes2pick: 20


# Plotting configuration
skeleton:
- - bodypart1
  - bodypart2
- - objectA
  - bodypart3
skeleton_color: black
pcutoff: 0.6
dotsize: 12
alphavalue: 0.7
colormap: rainbow


# Training,Evaluation and Analysis configuration
TrainingFraction:
- 0.95
iteration: 0
default_net_type: resnet_50
default_augmenter: default
snapshotindex: -1
detector_snapshotindex:
batch_size: 8


# Cropping Parameters (for analysis and outlier frame detection)
cropping: false
#if cropping is true for analysis, then set the values here:
x1: 0
x2: 640
y1: 277
y2: 624


# Refinement configuration (parameters from annotation dataset configuration also relevant in this stage)
corner2move2:
- 50
- 50
move2corner: true


# Conversion tables to fine-tune SuperAnimal weights
SuperAnimalConversionTables:
```

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by n-poulsen** [comment]
Hi @mikeumo! It seems there's an issue when loading the pretrained weights from ImageNet when training the network. Could you run `pip list` in your terminal (with your DeepLabCut conda environment activated) and report the output?

**Comment by mikeumo** [comment]
here you go...

```
Package                       Version
----------------------------- -----------
alabaster                     0.7.16
albumentations                1.4.3
app-model                     0.2.7
appdirs                       1.4.4
appnope                       0.1.4
asttokens                     2.4.1
attrs                         23.2.0
Babel                         2.15.0
build                         1.2.1
cachey                        0.2.1
certifi                       2024.6.2
charset-normalizer            3.3.2
click                         8.1.7
cloudpickle                   3.0.0
comm                          0.2.2
contourpy                     1.2.1
cycler                        0.12.1
dask                          2024.6.0
dask-expr                     1.1.3
dask-image                    2024.5.3
debugpy                       1.8.1
decorator                     5.1.1
deeplabcut                    3.0.0rc1
dlclibrary                    0.0.6
docker-pycreds                0.4.0
docstring_parser              0.16
docutils                      0.17.1
einops                        0.8.0
exceptiongroup                1.2.1
executing                     2.0.1
filelock                      3.13.1
filterpy                      1.4.5
flexcache                     0.3
flexparser                    0.3.1
fonttools                     4.53.0
freetype-py                   2.4.0
fsspec                        2024.2.0
gitdb                         4.0.11
GitPython                     3.1.43
HeapDict                      1.0.1
hsluv                         5.0.4
huggingface-hub               0.23.4
idna                          3.7
imageio                       2.34.1
imageio-ffmpeg                0.5.1
imagesize                     1.4.1
imgaug                        0.4.0
importlib_metadata            7.1.0
in-n-out                      0.2.1
ipykernel                     6.29.4
ipython                       8.25.0
jedi                          0.19.1
Jinja2                        3.1.3
joblib                        1.4.2
jsonschema                    4.22.0
jsonschema-specifications     2023.12.1
jupyter_client                8.6.2
jupyter_core                  5.7.2
kiwisolver                    1.4.5
lazy_loader                   0.4
llvmlite                      0.43.0
locket                        1.0.0
magicgui                      0.8.3
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
matplotlib                    3.8.4
matplotlib-inline             0.1.7
mdurl                         0.1.2
mpmath                        1.3.0
napari                        0.4.18
napari-console                0.0.9
napari-deeplabcut             0.2.1.7
napari-plugin-engine          0.2.0
napari-svg                    0.1.10
natsort                       8.4.0
nest-asyncio                  1.6.0
networkx                      3.2.1
npe2                          0.7.5
numba                         0.60.0
numexpr                       2.10.0
numpy                         1.26.3
numpydoc                      1.5.0
opencv-python                 4.10.0.84
opencv-python-headless        4.10.0.84
packaging                     24.1
pandas                        2.2.2
parso                         0.8.4
partd                         1.4.2
patsy                         0.5.6
pexpect                       4.9.0
pillow                        10.2.0
PIMS                          0.7
Pint                          0.24
pip                           24.0
platformdirs                  4.2.2
pooch                         1.8.2
prompt_toolkit                3.0.47
protobuf                      5.27.1
psutil                        6.0.0
psygnal                       0.11.1
ptyprocess                    0.7.0
pure-eval                     0.2.2
py-cpuinfo                    9.0.0
pyarrow                       16.1.0
pycocotools                   2.0.8
pyconify                      0.1.6
pydantic                      1.10.16
pydantic-compat               0.1.2
Pygments                      2.18.0
PyOpenGL                      3.1.7
pyparsing                     3.1.2
pyproject_hooks               1.1.0
PySide6                       6.4.2
PySide6-Addons                6.4.2
PySide6-Essentials            6.4.2
python-dateutil               2.9.0.post0
pytz                          2024.1
PyYAML                        6.0.1
pyzmq                         26.0.3
QDarkStyle                    3.1
qtconsole                     5.5.2
QtPy                          2.4.1
referencing                   0.35.1
requests                      2.32.3
rich                          13.7.1
rpds-py                       0.18.1
ruamel.yaml                   0.18.6
ruamel.yaml.clib              0.2.8
safetensors                   0.4.3
scikit-image                  0.24.0
scikit-learn                  1.5.0
scipy                         1.10.1
sentry-sdk                    2.5.1
setproctitle                  1.3.3
setuptools                    69.5.1
shapely                       2.0.4
shellingham                   1.5.4
shiboken6                     6.4.2
six                           1.16.0
slicerator                    1.1.0
smmap                         5.0.1
snowballstemmer               2.2.0
Sphinx                        4.5.0
sphinxcontrib-applehelp       1.0.8
sphinxcontrib-devhelp         1.0.6
sphinxcontrib-htmlhelp        2.0.5
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.7
sphinxcontrib-serializinghtml 1.1.10
stack-data                    0.6.3
statsmodels                   0.14.2
superqt                       0.6.7
sympy                         1.12
tables                        3.8.0
threadpoolctl                 3.5.0
tifffile                      2024.6.18
timm                          1.0.3
tomli                         2.0.1
tomli_w                       1.0.0
toolz                         0.12.1
torch                         2.3.1
torchaudio                    2.3.1
torchvision                   0.18.1
tornado                       6.4.1
tqdm                          4.66.4
traitlets                     5.14.3
typer                         0.12.3
typing_extensions             4.9.0
tzdata                        2024.1
urllib3                       2.2.2
vispy                         0.12.2
wandb                         0.17.2
wcwidth                       0.2.13
wheel                         0.43.0
wrapt                         1.16.0
zipp                          3.19.
```

**Comment by sravsrav** [comment]
I am also having the same error: 
64GB RAM M3 Pro

```
In [3]: deeplabcut.train_network(config_path, shuffle=1, displayiters=100, savei
   ...: ters=15000, maxiters=30000,allow_growth=True)
Training with configuration:
data:
  colormode: RGB
  inference:
    normalize_images: True
  train:
    affine:
      p: 0.5
      rotation: 30
      scaling: [1.0, 1.0]
      translation: 0
    collate:
      type: ResizeFromDataSizeCollate
      min_scale: 0.4
      max_scale: 1.0
      min_short_side: 128
      max_short_side: 1152
      multiple_of: 32
      to_square: False
    covering: False
    gaussian_noise: 12.75
    hist_eq: False
    motion_blur: False
    normalize_images: True
device: auto
metadata:
  project_path: /Users/sravani/Documents/DLC/SwimmingV1-Sravani-2024-06-25
  pose_config_path: /Users/sravani/Documents/DLC/SwimmingV1-Sravani-2024-06-25/dlc-models-pytorch/iteration-0/SwimmingV1Jun25-trainset95shuffle1/train/pose_cfg.yaml
  bodyparts: ['head', 'neck', 'chest', 'R elbow', 'R wrist', 'L elbow', 'L wrist', 'waist', 'R knee', 'R ankle', 'L knee', 'L ankle']
  unique_bodyparts: []
  individuals: ['animal']
  with_identity: None
method: bu
model:
  backbone:
    type: ResNet
    model_name: resnet50_gn
    output_stride: 16
    freeze_bn_stats: True
    freeze_bn_weights: False
  backbone_output_channels: 2048
  heads:
    bodypart:
      type: HeatmapHead
      weight_init: normal
      predictor:
        type: HeatmapPredictor
        apply_sigmoid: False
        clip_scores: True
        location_refinement: True
        locref_std: 7.2801
      target_generator:
        type: HeatmapGaussianGenerator
        num_heatmaps: 12
        pos_dist_thresh: 17
        heatmap_mode: KEYPOINT
        generate_locref: True
        locref_std: 7.2801
      criterion:
        heatmap:
          type: WeightedMSECriterion
          weight: 1.0
        locref:
          type: WeightedHuberCriterion
          weight: 0.05
      heatmap_config:
        channels: [2048, 12]
        kernel_size: [3]
        strides: [2]
      locref_config:
        channels: [2048, 24]
        kernel_size: [3]
        strides: [2]
net_type: resnet_50
runner:
  type: PoseTrainingRunner
  gpus: None
  key_metric: test.mAP
  key_metric_asc: True
  eval_interval: 1
  optimizer:
    type: AdamW
    params:
      lr: 0.0001
  scheduler:
    type: LRListScheduler
    params:
      lr_list: [[1e-05], [1e-06]]
      milestones: [160, 190]
  snapshots:
    max_snapshots: 5
    save_epochs: 25
    save_optimizer_state: False
train_settings:
  batch_size: 1
  dataloader_workers: 0
  dataloader_pin_memory: True
  display_iters: 100
  epochs: 200
  seed: 42
Loading pretrained weights from Hugging Face hub (timm/resnet50_gn.a1h_in1k)
[timm/resnet50_gn.a1h_in1k] Safe alternative available for 'pytorch_model.bin' (as 'model.safetensors'). Loading weights using safetensors.
zsh: segmentation fault  ipython
```

**Comment by n-poulsen** [comment]
@mikeumo and @sravsrav - it seems like the segfault occurs before training starts (likely when the weights are being downloaded). Could you try running the following code and let me know if it segfaults as well (you can do that in `ipython`):

```python
import timm
resnet = timm.create_model("resnet50_gn", output_stride=16, pretrained=True)
```

**Comment by mikeumo** [comment]
I don't get any errors. 

```
resnet
Out[3]: 
ResNet(
  (conv1): Conv2d(3, 64, kernel_size=(7, 7), stride=(2, 2), padding=(3, 3), bias=False)
  (bn1): GroupNorm(32, 64, eps=1e-05, affine=True)
  (act1): ReLU(inplace=True)
  (maxpool): MaxPool2d(kernel_size=3, stride=2, padding=1, dilation=1, ceil_mode=False)
  (layer1): Sequential(
    (0): Bottleneck(
      (conv1): Conv2d(64, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 64, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 64, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
      (downsample): Sequential(
        (0): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (1): GroupNorm(32, 256, eps=1e-05, affine=True)
      )
    )
    (1): Bottleneck(
      (conv1): Conv2d(256, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 64, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 64, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (2): Bottleneck(
      (conv1): Conv2d(256, 64, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 64, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(64, 64, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 64, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(64, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
  )
  (layer2): Sequential(
    (0): Bottleneck(
      (conv1): Conv2d(256, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 128, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 128, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(128, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
      (downsample): Sequential(
        (0): Conv2d(256, 512, kernel_size=(1, 1), stride=(2, 2), bias=False)
        (1): GroupNorm(32, 512, eps=1e-05, affine=True)
      )
    )
    (1): Bottleneck(
      (conv1): Conv2d(512, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 128, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 128, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(128, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (2): Bottleneck(
      (conv1): Conv2d(512, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 128, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 128, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(128, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (3): Bottleneck(
      (conv1): Conv2d(512, 128, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 128, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(128, 128, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 128, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(128, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
  )
  (layer3): Sequential(
    (0): Bottleneck(
      (conv1): Conv2d(512, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(2, 2), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
      (downsample): Sequential(
        (0): Conv2d(512, 1024, kernel_size=(1, 1), stride=(2, 2), bias=False)
        (1): GroupNorm(32, 1024, eps=1e-05, affine=True)
      )
    )
    (1): Bottleneck(
      (conv1): Conv2d(1024, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (2): Bottleneck(
      (conv1): Conv2d(1024, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (3): Bottleneck(
      (conv1): Conv2d(1024, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (4): Bottleneck(
      (conv1): Conv2d(1024, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (5): Bottleneck(
      (conv1): Conv2d(1024, 256, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 256, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(256, 256, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 256, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(256, 1024, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 1024, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
  )
  (layer4): Sequential(
    (0): Bottleneck(
      (conv1): Conv2d(1024, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(1, 1), bias=False)
      (bn2): GroupNorm(32, 512, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(512, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 2048, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
      (downsample): Sequential(
        (0): Conv2d(1024, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
        (1): GroupNorm(32, 2048, eps=1e-05, affine=True)
      )
    )
    (1): Bottleneck(
      (conv1): Conv2d(2048, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(2, 2), dilation=(2, 2), bias=False)
      (bn2): GroupNorm(32, 512, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(512, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 2048, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
    (2): Bottleneck(
      (conv1): Conv2d(2048, 512, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn1): GroupNorm(32, 512, eps=1e-05, affine=True)
      (act1): ReLU(inplace=True)
      (conv2): Conv2d(512, 512, kernel_size=(3, 3), stride=(1, 1), padding=(2, 2), dilation=(2, 2), bias=False)
      (bn2): GroupNorm(32, 512, eps=1e-05, affine=True)
      (drop_block): Identity()
      (act2): ReLU(inplace=True)
      (aa): Identity()
      (conv3): Conv2d(512, 2048, kernel_size=(1, 1), stride=(1, 1), bias=False)
      (bn3): GroupNorm(32, 2048, eps=1e-05, affine=True)
      (act3): ReLU(inplace=True)
    )
  )
  (global_pool): SelectAdaptivePool2d(pool_type=avg, flatten=Flatten(start_dim=1, end_dim=-1))
  (fc): Linear(in_features=2048, out_features=1000, bias=True)
)

In [4]: 
```

**Comment by n-poulsen** [comment]
Thanks - I'll continue trying to replicate the issue

**Comment by mikeumo** [comment]
I created a (mini)conda environment with python=3.11, installed conda-forge::pytables=3.8.0 and conda-forge::gfortran (not sure if I needed to),  activated the environment and installed DeepLabCut from git with 

pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]"

Started the DLC, used the project that was crushing before, but now I am able to train the network! It is training at the moment and giving me the performance starts by epoch in the terminal.

Just FYI.

Also, I tried to use python=3.12 by the scipy-1.9 is not working on python 3.12. 

Is it possible to update the scipy dependency in DLC to 1.14?

**Comment by phragmosis** [comment]
Identical issue here as well (M3 Max with Sonoma 14.5)

**Comment by n-poulsen** [comment]
@mikeumo Currently we only support python 3.10 and 3.11 (because some dependencies are not yet compatible with 3.12). I'll look into upgrading the `scipy` version, but we pinned the version as their new releases were making tests fail.

As a temporary fix (while we work on a solution for MPS), you should be able to get around the segfault by training on CPU (set `device="cpu"` when calling API methods or update the model configuration file to have `device: cpu`).

**Comment by tom21100227** [comment]
Same issue here (M2 Macbook Pro with Sonoma 14.5) on `3.0.0rc2`. Python version 3.10.14: Both the command line option (`deeplabcut.train_network()`) and the gui have the same issue: 
<img width="736" alt="Screenshot 2024-07-06 at 18 55 37" src="https://github.com/DeepLabCut/DeepLabCut/assets/103209542/34adad21-2edd-4538-9f1d-ef69813fd29a">

**Comment by mikeumo** [comment]
With python=3.11 I can bypass the crash before training, but when I get to analyze videos, all I get is a bunch of empty plots. Am I missing something?

**Comment by tom21100227** [comment]
> I created a (mini)conda environment with python=3.11, installed conda-forge::pytables=3.8.0 and conda-forge::gfortran (not sure if I needed to), activated the environment and installed DeepLabCut from git with
> 
> pip install "git+[https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]](https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut%5Bgui,modelzoo,wandb%5D)"
> 
> Started the DLC, used the project that was crushing before, but now I am able to train the network! It is training at the moment and giving me the performance starts by epoch in the terminal.
> 
> Just FYI.
> 
> Also, I tried to use python=3.12 by the scipy-1.9 is not working on python 3.12.
> 
> Is it possible to update the scipy dependency in DLC to 1.14?

@mikeumo Hi, I tried to replicate your solution with the following command and I couldn't get DLC to train and still receive the same segfault. Did I missed something? 

```bash
conda create -n dlc_env python=3.11
conda activate dlc_env
conda install -c conda-forge pytables=3.8.0
conda install -c conda-forge gfortran
pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]"
# Go to deeplabcut and run train_network()`
```

**Comment by mikeumo** [comment]
I install the pytables and gfortran from outside of the environment as:

conda create --name dlc3 python=3.11
conda install --name dlc3 conda-forge::gfortran  
conda install --name dlc3 conda-forge::pytables=3.8.0

then, after I active the dlc3 environment, 

conda activate dlc3


i install the DLC from git:

pip install "git+https://github.com/DeepLabCut/DeepLabCut.git@pytorch_dlc#egg=deeplabcut[gui,modelzoo,wandb]"

(so, I install the deeplabcut only from within the dlc3 environment, just to be clear. NOT pytables or gfortran)
Then, just run the DLC:

python -m deeplabcut


My machine is 

M1 Ultra (Mac Studio), 64 gb RAM, Sonoma 14.5

So, I use miniconda.

I also tried the same install sequence on M2 Mac and I can train the network there as well. But I still get empty plots in Analyze Videos stage.

**Comment by tom21100227** [comment]
@mikeumo Hi Mike, thank you for the fix! I was able to get deeplabcut running with your fix, and I was able to see all the plots in `plot-poses/*` directory of my original video. I am on an M2 MacBook Pro, 16GB of RAM and Sonoma 14.5 as well. Did you check the "plot-trajectory` options on when running analyze? Happy to provide more detail about my DLC environment/dependency versions in case you want to troubleshoot more.
